### ISSUE TITLE
Allow user to choose to always run a2 as admin?

### ISSUE THREAD

**Issue by lipkau** [ewerybody/a2#117]
### Expected Behavior
User should be able to tell a2 to always start as elevated.
This can be required by modules.

### Current Behavior
User is unable to choose this

### Possible Solution
add a control to the a2ui, where the user can choose if a2 should be started as admin or not.
in a2.ahk:
```
if (!A_IsAdmin && Setting.RunAsAdmin == "Always/Ask")
    RunAsAdmin()

RunAsAdmin()
{
    global
    local params, uacrep
    Loop %0%
        params .= " " (InStr(%A_Index%, " ") ? """" %A_Index% """" : %A_Index%)
    if (A_IsCompiled)
        uacrep := DllCall("shell32\ShellExecute", uint, 0, str, "RunAs", str, A_ScriptFullPath, str, "/r" params, str, A_WorkingDir, int, 1)
    else
        uacrep := DllCall("shell32\ShellExecute", uint, 0, str, "RunAs", str, A_AhkPath, str, "/r """ A_ScriptFullPath """" params, str, A_WorkingDir, int, 1)
    if (uacrep = 42) ;UAC Prompt confirmed, application may run as admin
        ExitApp
    else
        MsgBox 7plus is running in non-admin mode. Some features will not be working.
}
```

**Comment by ewerybody** [comment]
Ah yeah that could be useful! Will Windows then always ask to Execute it as Admin? Like when you run RegEdit or such?

**Comment by lipkau** [comment]
not necessarily.
this is called UAC. If the user changed the value in the windows' control panel, the prompt will not be shown

**Comment by ewerybody** [comment]
That might even be a thing for multi process a2. So that only the modules that actually need it receive elevation.

**Comment by lipkau** [comment]
but we are not there yet!

**Comment by ewerybody** [comment]
yet! ☝🏼 😄

**Comment by lipkau** [comment]
This can be implemented with #119

**Comment by ewerybody** [comment]
backlogging this for now. I would actually not advise to run any AHK as Admin really.
